### ISSUE TITLE
4D/BTi import error if pdf/config channel orders to not match

### ISSUE THREAD

**Issue by joshbear** [mne-tools/mne-python#6032]
#### Describe the bug
If the order of channels is different in the config and header (pdf) files, than the current match algorithm will report that they do not match even if all channels are present in both. 

#### Steps and/or code to reproduce
You need to have a config and header file with channels in a different order. If you do, then all you need to do is:

raw = mne.io.read_raw_bti(filename) 

#### Expected results
import the file

#### Actual results
throws a runtime exception

#### Additional information
Platform:      Darwin-18.2.0-x86_64-i386-64bit
Python:        3.6.5 |Anaconda, Inc.| (default, Apr 26 2018, 08:42:37)  [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
Executable:    /Users/joshbear/bin/anaconda3/envs/mne/bin/python
CPU:           i386: 8 cores
Memory:        16.0 GB

mne:           0.17.0
numpy:         1.14.2 {blas=mkl_rt, lapack=mkl_rt}
scipy:         1.1.0
matplotlib:    2.2.2 {backend=MacOSX}

sklearn:       0.19.1
nibabel:       2.2.1
mayavi:        4.5.1.dev0 {qt_api=pyqt5, PyQt5=5.10.1}
cupy:          Not found
pandas:        0.23.0
dipy:          Not found

**Comment by agramfort** [comment]
maybe @pravsripad or @jdammers can help?

**Comment by jdammers** [comment]
@joshbear would you mind sharing the files for testing?

**Comment by joshbear** [comment]
@jdammers Try this link: https://1drv.ms/u/s!AjNKUsqPwOwNgVQU6qT5fnX4lyXt
It's about 75 mb. 10 minute recording. Let me know if problems.

**Comment by jdammers** [comment]
@joshbear I can confirm that the number of channels found is the same in the pdf and the config. However, as you mentioned, the channel order in the pdf is different (kind of unsorted) as compared to the config (channels are sorted). The current implementation of read_raw_bti() compares each entry of the pdf channel list with the channel list in the config file (i.e., a 1-by-1 matching) and therefore it complains about the unmatch.  What is the reason for the different channel order in the pdf?

**Comment by joshbear** [comment]
Honestly, I'm not sure. I've tried to figure that out from our physicist, but his initial impression was that it is just a quirk in MNE. However, I've processed a couple dozen files, and this one run from this one subject is the only one that seemed off. Indeed, I've recently (pleasantly) discovered that the other runs from the same recording don't have this problem. It could be a one-off quirk, and I can discard this run (for my current purposes, it won't really matter that much — when I thought it was affecting all runs, it was a bigger problem). but it will raises the question of whether there is some way that the channels could be properly re-ordered without introducing new errors?
